src/lsm/grid.zig

```diff
@@ -170,17 +170,21 @@ pub fn GridType(comptime Storage: type) type {
             // Resolve reads that were seen in the cache during start_read()
             // but deferred to be asynchronously resolved on the next tick.
             //
-            // Make a copy to avoid a potential infinite loop here.
-            // If any new reads are added to the read_cached_queue they will be started on
-            // the next tick.
-            var copy = grid.read_cached_queue;
-            grid.read_cached_queue = .{};
-            while (copy.pop()) |read| {
+            // Drains directly from the queue to that cache reads that could be serviced immediately
+            // aren't deferred until the next tick (which could be milliseconds from IO.run_for_ns).
+            //
+            // Even still, there should still be a cap on reads processed to prevent going over
+            // any implicit time slice expected of Grid.tick(). This limit is fairly arbitrary.
+            var max_retry: u32 = Grid.read_iops_max * 4;
```

Comment:
This is only `15*4 = 60` cache reads, which I think isn't enough to resolve https://github.com/tigerbeetledb/tigerbeetle/issues/226. I understand that we don't want Grid to depend on StateMachine to compute the upper bound, but it would come out to approximately:

```
( 8191 debit accounts
+ 8191 credit accounts
+ 8191 pending transfers
+ 8191 posting transfers ) * (1 index block + 1 filter block + 1 data block) = 98292
```

Is there a downside to setting `max_ retry` to a value that high? i.e. it might delay our next IO tick (but not too long I wouldn't think), but why is that an issue here?

Reply:
I wasn't actually sure how high it should go. Without reaching into state_machine, what would be an ok upper bound (or should we have one at all)? Spending too much time in `Grid.tick()` could affect the ticking of the ping/view_change/etc. timeouts in the Replica and I'm not sure how much of an effect that could have.